sm/tedge_agent/src/main.rs

```diff
@@ -19,13 +20,18 @@ async fn main() -> Result<(), anyhow::Error> {
     initialise_logging();
 
     let default_plugin_type = tedge_config.query_string_optional(SoftwarePluginDefaultSetting)?;
+    let mqtt_config =
+        mqtt_client::Config::default().with_port(tedge_config.query(MqttPortSetting)?.into());
+
     let tedge_config_path = config_repository
         .get_config_location()
         .tedge_config_root_path()
         .to_path_buf();
     let sm_agent_config = SmAgentConfig::default()
         .with_default_plugin_type(default_plugin_type)
-        .with_sm_home(tedge_config_path);
+        .with_sm_home(tedge_config_path)
+        .with_mqtt_client_config(mqtt_config);
+
```

Comment:
Can you please move all this code in a `SmAgentConfig::new(tedge_config_location)` method taking a tedge config as an argument.